### PR TITLE
Saul 47

### PR DESCRIPTION
**Pull request by NahuelBerg**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: azure-cli2017.pyproj [hunk 1]
```diff
@@ -13,8 +13,7 @@
     <LaunchProvider>Standard Python launcher</LaunchProvider>
     <InterpreterId>MSBuild|{54f4b6dc-0859-46dc-99bb-b275c9d0aca3}|$(MSBuildProjectFullPath)</InterpreterId>
     <EnableNativeCodeDebugging>False</EnableNativeCodeDebugging>
-    <CommandLineArguments>
-    </CommandLineArguments>
+    <CommandLineArguments>ams content-key-policy create --policy-option-name policyOption --issuer Issuer --audience Audience --restriction-token-type Jwt -g test-fbesteiro -a fbesteiro -n testPolicyDemo --widevine-template "C:/widevineTemplate.json" --symmetric-token-key someTokenDesc</CommandLineArguments>
```
Comment (Renkon) [inline azure-cli2017.pyproj]: remove custom setting

File: src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/_params.py [hunk 2]
```diff
@@ -161,6 +161,7 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
         c.argument('restriction_token_type', arg_group='Token Restriction Parameters',
                    arg_type=get_enum_type(ContentKeyPolicyRestrictionTokenType), help='The type of token.')
         c.argument('open_id_connect_discovery_document', arg_group='Token Restriction Parameters', help='The OpenID connect discovery document.')
+        c.argument('widevine_template', arg_group='Token Configuration Parameters', help='The Widevine template.')
```
Comment (Renkon) [inline src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/_params.py]: You should not really create a custom group given its a single command.

File: src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/operations/content_key_policy.py [hunk 3]
```diff
@@ -89,8 +91,11 @@ def _generate_content_key_policy_object(client, resource_group_name, account_nam
     if open_restriction and valid_token_restriction:
         raise CLIError('You should only use one restriction type.')
 
-    if clear_key_configuration:
+    if clear_key_configuration and not widevine_template:
```
Comment (Renkon) [inline src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/operations/content_key_policy.py]: You should first check for both. if both are true then raise an error

File: src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/operations/content_key_policy.py [hunk 4]
```diff
@@ -89,8 +91,11 @@ def _generate_content_key_policy_object(client, resource_group_name, account_nam
...
         configuration = ContentKeyPolicyClearKeyConfiguration()
+    elif widevine_template:
```
Comment (Renkon) [inline src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/operations/content_key_policy.py]: leave an empty line between

File: src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/operations/content_key_policy.py [hunk 5]
```diff
@@ -132,10 +137,11 @@ def _generate_content_key_policy_object(client, resource_group_name, account_nam
         for key in _x509_keys:
             alternative_keys.append(_x509_token_key_factory(key))
 
-        for key in token_claims:
-            claim = ContentKeyPolicyTokenClaim(claim_type=key,
-                                               claim_value=token_claims[key])
-            _token_claims.append(claim)
+        if token_claims is not None:
```
Comment (Renkon) [inline src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/operations/content_key_policy.py:144]: good catch (y)

**Review by Renkon — CHANGES_REQUESTED** [review]

File: src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/_params.py (hunk 2 above)
Comment (NahuelBerg) [reply src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/_params.py]: Yes, now it's a single configuration option, there will be more in this iteration, that's why I created a new arg_group

File: src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/operations/content_key_policy.py (hunk 4 above)
Comment (NahuelBerg) [reply src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/operations/content_key_policy.py]: I don't get this. Why would I need to leave an empty line there?

**Review by Renkon — COMMENTED** [review]
lgtm

**Review by Renkon — APPROVED** [review]

**Review by damatojuan — APPROVED** [review]
lgtm

**Pull request merged**